ospf6d/ospf6_top.c

```diff
@@ -865,88 +866,203 @@ DEFUN (no_ospf6_stub_router_shutdown,
 }
 #endif
 
-static void ospf6_show(struct vty *vty, struct ospf6 *o)
+
+static void ospf6_show(struct vty *vty, struct ospf6 *o, json_object *json,
+		       bool use_json)
 {
 	struct listnode *n;
 	struct ospf6_area *oa;
 	char router_id[16], duration[32];
 	struct timeval now, running, result;
 	char buf[32], rbuf[32];
+	json_object *json_areas = NULL;
+
+	if (use_json) {
+		json_areas = json_object_new_object();
+
+		/* process id, router id */
+		inet_ntop(AF_INET, &o->router_id, router_id, sizeof(router_id));
+		json_object_string_add(json, "routerId", router_id);
+
+		/* running time */
+		monotime(&now);
+		timersub(&now, &o->starttime, &running);
+		timerstring(&running, duration, sizeof(duration));
+		json_object_string_add(json, "running", duration);
+
+		/* Redistribute configuration */
+		/* XXX */
+		json_object_int_add(json, "lsaMinimumArrivalMsecs",
+				    o->lsa_minarrival);
+
+		/* Show SPF parameters */
+		json_object_int_add(json, "spfScheduleDelayMsecs",
+				    o->spf_delay);
+		json_object_int_add(json, "holdTimeMinMsecs", o->spf_holdtime);
+		json_object_int_add(json, "holdTimeMaxMsecs",
+				    o->spf_max_holdtime);
+		json_object_int_add(json, "holdTimeMultplier",
+				    o->spf_hold_multiplier);
+
+
+		if (o->ts_spf.tv_sec || o->ts_spf.tv_usec) {
+			timersub(&now, &o->ts_spf, &result);
+			timerstring(&result, buf, sizeof(buf));
+			ospf6_spf_reason_string(o->last_spf_reason, rbuf,
+						sizeof(rbuf));
+			json_object_string_add(json, "spfLastExecutedMsecs",
+					       buf);
+			json_object_string_add(json, "spfLastExecutedReason",
+					       rbuf);
+
+			json_object_int_add(
+				json, "spfLastDurationSecs",
+				(long long)o->ts_spf_duration.tv_sec);
+
+			json_object_int_add(
+				json, "spfLastDurationMsecs",
+				(long long)o->ts_spf_duration.tv_usec);
+		} else
+			json_object_boolean_true_add(json, "spfHasNotRun");
+
+
+		threadtimer_string(now, o->t_spf_calc, buf, sizeof(buf));
+		if (o->t_spf_calc) {
+			long time_store;
+
+			json_object_boolean_true_add(json, "spfTimerActive");
+			time_store =
+				monotime_until(&o->t_spf_calc->u.sands, NULL)
+				/ 1000LL;
+			json_object_int_add(json, "spfTimerDueInMsecs",
+					    time_store);
+		} else
+			json_object_boolean_false_add(json, "spfTimerActive");
+
+		if (CHECK_FLAG(o->flag, OSPF6_STUB_ROUTER))
+			json_object_boolean_true_add(json,
+						     "routerIsStubRouter");
+
+		/* LSAs */
+		json_object_int_add(json, "numberOfAsScopedLsa",
+				    o->lsdb->count);
+		/* Areas */
+		json_object_int_add(json, "numberOfAreaInRouter",
+				    listcount(o->area_list));
+		if (CHECK_FLAG(o->config_flags, OSPF6_LOG_ADJACENCY_CHANGES)) {
+			if (CHECK_FLAG(o->config_flags,
+				       OSPF6_LOG_ADJACENCY_DETAIL))
+				json_object_boolean_true_add(
+					json, "adjacencyChangesLoggedAll");
+			else
+				json_object_boolean_true_add(
+					json, "adjacencyChangesLogged");
```

Comment:
As before. Rather than conditionally including a key, why not make it a single key with a specifier value, like:
```
adjacencyChanges: "logged" | "loggedAll"
```
etc